small improvement for _tmp directory i/o

raffazizzi opened the pull request:
writing to the `_tmp` folder was causing some issues under certain environments (e.g. webpack including a typescript compilation) so I switched to a more robust approach using `path.join` and `url.fileURLToPath`

raffazizzi committed:
small improvement for _tmp directory i/o

joeytakeda commented:
Thanks for this, @raffazizzi ! This is much improved, but when I tried this out, the tests failed due to the now deprecated use of import assertions. 

I've now fixed that, but in doing so, there's now a conflict with the `src/stylesheets/sef` directory. Those files should probably not be kept in the repository anyway (since they should just be generated postinstall anyway), but for the time being, could you update your fork and make sure all still works for you? Once that's done, I'll merge and tag a new release.

raffazizzi committed:
resolved sef conflicts

raffazizzi committed:
removed sef artifacts; added xslt3 dependency

raffazizzi committed:
wrapped `exec` in _transformSVRLtoSEF() with a promise to make sure that _tmp artifacts are created correctly before continuing validation process.

raffazizzi commented:
Thanks @joeytakeda! I made a coupe more changes:
* added `src/stylesheets/sef` to `.gitignore` and removed existing artifacts.
* added dev dependency to `xslt3`. I didn't have it installed globally and that caused the postinstall and tests to fail
* wrapped `exec` in _transformSVRLtoSEF() with a Promise because it was taking long to run on my machine and the code would move on too quickly expecting `/src/_tmp/svrl.sef.json` to have already been generated. This shouldn't change anything in practice since _transformSVRLtoSEF() is already `async` and waiting on other processes

raffazizzi commented:
Scratch the second bullet point about `xstl3` – not sure what happened there since it was already a dependency. I moved it back as a dependency in package.json.

raffazizzi committed:
moved xslt3 back to regular dependency

joeytakeda commented:
Excellent! Thanks very much, @raffazizzi ! Will merge now.
